components/ws-daemon/pkg/controller/workspace_operations.go

```diff
@@ -290,6 +298,51 @@ func (wso *DefaultWorkspaceOperations) DeleteWorkspace(ctx context.Context, inst
 	return nil
 }
 
+func (wso *DefaultWorkspaceOperations) WipeWorkspace(ctx context.Context, instanceID string) error {
+	log := log.New().WithContext(ctx)
+
+	ws, err := wso.provider.GetAndConnect(ctx, instanceID)
+	if err != nil {
+		// we have to assume everything is fine, and this workspace has already been completely wiped
+		return nil
+	}
+	log = log.WithFields(ws.OWI())
+
+	// mark this session as being wiped
+	ws.DoWipe = true
+
+	if err = ws.Dispose(ctx, wso.provider.hooks[session.WorkspaceDisposed]); err != nil {
+		log.WithError(err).Error("cannot dispose session")
+		return err
+	}
+
+	// dispose all running "dispatch handlers", e.g. all code running on the "pod informer"-triggered part of ws-daemon
+	wso.dispatch.DisposeWorkspace(ctx, instanceID)
+
+	// remove workspace daemon directory in the node
+	removedChan := make(chan struct{}, 1)
+	go func() {
+		defer close(removedChan)
+
+		if err := os.RemoveAll(ws.ServiceLocDaemon); err != nil {
+			log.WithError(err).Warn("cannot delete workspace daemon directory, leaving it dangling...")
+		}
+	}()
+
+	timeout := time.NewTicker(10 * time.Second)
```

Comment:
Added a comment.

tl;dr: If we don't have the timeout, there is a chance that that this operation blocks forever (at least before I fixed the mark unmounting). By adding the timeout guarantee it never blocks the workspace from being deleted (because that depends on the "wiping" to finish.

We trade it for the slight chance of not wiping cleanly, which _might_ become a problem iff the new workspace pod is scheduled to the same node again. Which I deem ok, because then we are talking about an error case in the 0.1-0.01% range, which is already way better than the original 1%.
All of this, again, only if the mark unmounting did not fix this problem in the first place. :+1: